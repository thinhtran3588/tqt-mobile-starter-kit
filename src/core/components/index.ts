export * from '@tqt/mobile/lib/components';
export * from './icon/icon.component';
export {
  Provider as PaperProvider,
  DefaultTheme as PaperDefaultTheme,
  Colors,
  Button,
  Appbar,
  Card,
  ActivityIndicator,
  Avatar,
  Badge,
} from 'react-native-paper';
export {StatusBar, Alert, ScrollView} from 'react-native';
