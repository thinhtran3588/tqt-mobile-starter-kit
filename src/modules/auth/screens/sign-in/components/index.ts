export * from './social-sign-in/social-sign-in.component';
