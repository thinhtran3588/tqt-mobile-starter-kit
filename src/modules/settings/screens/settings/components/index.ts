export * from './theme-setting/theme-setting.component';
export * from './primary-color-setting/primary-color-setting.component';
export * from './general-info/general-info.component';
export * from './language-setting/language-setting.component';
