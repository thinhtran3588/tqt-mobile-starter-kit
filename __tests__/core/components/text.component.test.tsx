import React from 'react';
import {render, toJSON} from '@test-utils';
import {Text} from '@core/components';

it('renders correctly', async () => {
  const {baseElement} = render(<Text>Text</Text>);
  expect(toJSON(baseElement)).toMatchSnapshot();
});

it('renders type correctly', async () => {
  const {baseElement} = render(<Text type='h1'>Text</Text>);
  expect(toJSON(baseElement)).toMatchSnapshot();
});

it('renders uppercase correctly', async () => {
  const {baseElement} = render(
    <Text type='h1' uppercase>
      Text
    </Text>,
  );
  expect(toJSON(baseElement)).toMatchSnapshot();
});

it('renders lowercase correctly', async () => {
  const {baseElement} = render(
    <Text type='h1' lowercase>
      Text
    </Text>,
  );
  expect(toJSON(baseElement)).toMatchSnapshot();
});

it('renders nested child correctly', async () => {
  const {baseElement} = render(
    <Text type='h1' lowercase>
      <Text>Text</Text>
    </Text>,
  );
  expect(toJSON(baseElement)).toMatchSnapshot();
});
