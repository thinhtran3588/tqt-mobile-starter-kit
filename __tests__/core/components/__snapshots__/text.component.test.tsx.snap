// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={false}
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Array [
              Object {
                "color": "#000000",
              },
              undefined,
              Object {
                "fontSize": 12,
              },
            ],
          ]
        }
      >
        Text
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders lowercase correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={false}
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Array [
              Object {
                "color": "#000000",
              },
              undefined,
              Object {
                "fontSize": 24,
                "fontWeight": "bold",
              },
            ],
          ]
        }
      >
        text
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders nested child correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={false}
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Array [
              Object {
                "color": "#000000",
              },
              undefined,
              Object {
                "fontSize": 24,
                "fontWeight": "bold",
              },
            ],
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "color": "#000000",
                },
                undefined,
                Object {
                  "fontSize": 12,
                },
              ],
            ]
          }
        >
          Text
        </Text>
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders type correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={false}
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Array [
              Object {
                "color": "#000000",
              },
              undefined,
              Object {
                "fontSize": 24,
                "fontWeight": "bold",
              },
            ],
          ]
        }
      >
        Text
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders uppercase correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={false}
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Array [
              Object {
                "color": "#000000",
              },
              undefined,
              Object {
                "fontSize": 24,
                "fontWeight": "bold",
              },
            ],
          ]
        }
      >
        TEXT
      </Text>
    </View>
  </View>
</View>
`;
